src/compiler/builders.js

```diff
@@ -13,12 +13,11 @@ const builderMap = {
 
 /**
  * Returns the set of component builders needed to server-render an AMP Document.
- *
- * @param {!./types.VersionsDef} versions
- * @return {Object<string, !./types.BuildDomDef>} builders
+ * @param {import('./types').Versions} versions
+ * @return {import('./types').BuilderMap}
  */
 export function getBuilders(versions) {
-  const builders = {};
+  const builders = /** @type {import('./types').BuilderMap} */ ({});
```

Comment:
Can you just import the types instead of this?

Reply:
not in a JS file

Reply:
It looks like that's because this is a JS file so it can't import from a TS file

Reply:
suggestion: type declaration rather than type-casting assignment
```suggestion
  /** @type {import('./types').BuilderMap} */
  const builders = {};
```